kong/plugins/file-log/handler.lua

```diff
@@ -47,13 +47,20 @@ local function log(premature, conf, message)
 
   local msg = cjson.encode(message).."\n"
 
-  local fd = get_fd(conf.path)
+  local fd 
+
+  if not conf.no_cache_fd then
+    fd = get_fd(conf.path)
+  end
+
```

Comment:
this does not close an existing FD if the config is changed dynamically. The existing one will remain in the cache. 

So if you set it to `false` it caches. Then set it to `true` it stop caching, but the old one remians in the cache. Logs now get rotated. Then customer sets it back to `false`. Now it reverts to the old, but by now stale, FD still in the cache.